compiler/optimizer/src/main/java/edu/snu/nemo/compiler/optimizer/pass/compiletime/reshaping/SailfishRelayReshapingPass.java

```diff
@@ -60,6 +62,21 @@ public SailfishRelayReshapingPass() {
             // before the vertex receiving shuffled data.
             final OperatorVertex iFileMergerVertex = new OperatorVertex(new RelayTransform());
             iFileMergerVertex.getExecutionProperties().put(SkipSerDesProperty.of());
+
+            // Update additional tagged output property of the source vertex if necessary.
+            if (edge.getSrc().getPropertyValue(AdditionalTagOutputProperty.class).isPresent()) {
+              final HashMap<String, String> additionalTagOutputPropertyValue =
+                  edge.getSrc().getPropertyValue(AdditionalTagOutputProperty.class).get();
+              if (additionalTagOutputPropertyValue.containsValue(v.getId())) {
+                additionalTagOutputPropertyValue.forEach((tag, vertexId) -> {
+                  if (vertexId == v.getId()) {
```

Comment:
I suggest `vertexId.equals(v.getId())`

Reply:
Thanks for your comments. I will fix this.